src/controllers/shared.controllers/director.residential.address.controller.ts

```diff
@@ -0,0 +1,138 @@
+          ${officerFiling.serviceAddress?.locality},
+          ${officerFiling.serviceAddress?.region ? officerFiling.serviceAddress.region+"," : ""}
+          `) + (officerFiling.serviceAddress?.country ? officerFiling.serviceAddress.country : "")
+          + officerFiling.serviceAddress?.postalCode
+};
+
+const formatDirectorRegisteredOfficeAddress = (companyProfile: CompanyProfile): string => {
+  return formatTitleCase(`
+          ${companyProfile.registeredOfficeAddress?.premises ? companyProfile.registeredOfficeAddress.premises+',' : ""}
+          ${companyProfile.registeredOfficeAddress?.addressLineOne},
+          ${companyProfile.registeredOfficeAddress?.addressLineTwo ? companyProfile.registeredOfficeAddress.addressLineTwo+',' : ""}
+          ${companyProfile.registeredOfficeAddress?.locality},
+          ${companyProfile.registeredOfficeAddress?.region ? companyProfile.registeredOfficeAddress.region+"," : ""}
+          ${companyProfile.registeredOfficeAddress?.country ? companyProfile.registeredOfficeAddress.country : ""}
+        `) + companyProfile.registeredOfficeAddress?.postalCode
+ }
+const checkRedirectUrl = (officerFiling: OfficerFiling, nextPageUrl: string, res: Response<any, Record<string, any>>, req: Request) => {
+  return officerFiling.checkYourAnswersLink && officerFiling.isServiceAddressSameAsRegisteredOfficeAddress
+    ? res.redirect(urlUtils.getUrlToPath(APPOINT_DIRECTOR_CHECK_ANSWERS_PATH, req))
+    : res.redirect(nextPageUrl);
+};
+
+const renderPage = (req: Request, res: Response, officerFiling: OfficerFiling, companyProfile: CompanyProfile, templateName: string, backUrlPath: string, formattedErrors?: FormattedValidationErrors) => {
+  const registeredOfficeAddress = mapCompanyProfileToOfficerFilingAddress(companyProfile.registeredOfficeAddress);
+  let canUseRegisteredOfficeAddress = false;
+  if (registeredOfficeAddress !== undefined) {
+      const registeredOfficeAddressAsCorrespondenceAddressErrors = validateManualAddress(registeredOfficeAddress, ResidentialManualAddressValidation);
+      if (registeredOfficeAddressAsCorrespondenceAddressErrors.length === 0) {
+        canUseRegisteredOfficeAddress = true;
+      }
+  }
+  logger.debug((canUseRegisteredOfficeAddress ? "Can" : "Can't") + " use registered office address copy for residential address");
+
+  return res.render(Templates.DIRECTOR_RESIDENTIAL_ADDRESS, {
+    templateName: templateName,
+    backLinkUrl: urlUtils.getUrlToPath(backUrlPath, req),
+    errors: formattedErrors,
+    director_address: officerFiling.directorResidentialAddressChoice,
+    directorName: formatTitleCase(retrieveDirectorNameFromFiling(officerFiling)),
+    directorRegisteredOfficeAddress: formatDirectorRegisteredOfficeAddress(companyProfile),
+    manualAddress: formatDirectorResidentialAddress(officerFiling),
+    protectedDetailsBackLink: DIRECTOR_RESIDENTIAL_ADDRESS_PATH_END,
```

Comment:
Do we need protectedDetailsBackLink in the template? Think we can remove?

Reply:
Have removed this now